charts/operator-wandb/templates/_bucket.tpl

```diff
@@ -30,18 +30,42 @@ secretKey: {{ $secretKey }}
 accessKeyName: {{ .Values.global.bucket.secret.accessKeyName }}
 secretKeyName: {{ .Values.global.bucket.secret.secretKeyName }}
 secretName: {{ include "wandb.bucket.secret" . }}
+{{- if eq $path "" -}}
+
+{{- if eq $provider "az" -}}
+{{- $url = "az://$(BUCKET_NAME)" -}}
+{{- end -}}
+
+{{- if eq $provider "gcs" -}}
+{{- $url = "gs://$(BUCKET_NAME)" -}}
+{{- end -}}
+
+{{- if eq $provider "s3" -}}
+{{- if or (and $accessKey $secretKey) .Values.global.bucket.secret.secretName -}}
+{{- $url = "s3://$(BUCKET_ACCESS_KEY):$(BUCKET_SECRET_KEY)@$(BUCKET_NAME)" -}}
```

Comment:
Won't this be a problem in the self-hosted s3 case where I believe we end up setting BUCKET_NAME to the url of the s3 server?

Reply:
I guess this is ok as path wouldn't be none in that case...